docs/aws.md

```diff
@@ -573,6 +573,55 @@ spark-sql --packages org.apache.iceberg:iceberg-spark3-runtime:{{% icebergVersio
     --conf spark.sql.catalog.my_catalog.client.assume-role.region=ap-northeast-1
 ```
 
+### URL Connection HTTP Client
```

Comment:
can we make this level 4, and at level 3 have a dedicated section for `HTTP client configurations`?

Reply:
And although it is just 1 line at this moment, can we also have a table for choosing the client type under the level 3 section? And we can talk a bit about the pros and cons of different HTTP client implementations

Reply:
Thank you for your suggestions. I refactored the doc to have a paragraph discuss how to choose between these two HTTP clients